storage/block/retriever_manager.go

```diff
@@ -102,10 +102,11 @@ func (r *shardBlockRetriever) Stream(
 	ctx context.Context,
 	id ident.ID,
 	blockStart time.Time,
+	blockEnd time.Time,
```

Comment:
As per other comment, hopefully we can avoid callers needing to pass blockEnd by resolving the block size on the retriever end.